--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,23 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+)
+
+cmake(
+    name = "apriltag",
+    cache_entries = {
+        "CMAKE_CXX_FLAGS": "-fPIC",
+        "CMAKE_BUILD_TYPE": "Release",
+        "BUILD_PYTHON_WRAPPER": "OFF",
+    },
+    copts = [
+        "-fPIC",
+        "-Wno-sign-compare",
+        "-Wno-unused-variable",
+    ],
+    lib_source = ":all_srcs",
+    out_shared_libs = ["libapriltag.so.3"],
+    visibility = ["//visibility:public"]
+)
