--- FreeRTOSv202112.00/BUILD	1969-12-31 17:00:00.000000000 -0700
+++ FreeRTOSv202112.00-modify/BUILD	2022-10-13 12:52:53.249513182 -0600
@@ -0,0 +1,45 @@
+cc_library(
+    name = "freertos",
+    srcs = glob([
+        "FreeRTOS/Source/*.c",
+    ]),
+    hdrs = glob(["FreeRTOS/Source/include/*.h"]),
+    include_prefix = "RTOS",
+    includes = ["FreeRTOS/Source/include"],
+    strip_include_prefix = "FreeRTOS/Source/include",
+)
+
+cc_library(
+    name = "heap1",
+    srcs = ["FreeRTOS/Source/portable/MemMang/heap_1.c"]
+)
+
+cc_library(
+    name = "heap2",
+    srcs = ["FreeRTOS/Source/portable/MemMang/heap_2.c"]
+)
+
+cc_library(
+    name = "heap3",
+    srcs = ["FreeRTOS/Source/portable/MemMang/heap_3.c"]
+)
+
+cc_library(
+    name = "heap4",
+    srcs = ["FreeRTOS/Source/portable/MemMang/heap_4.c"]
+)
+
+cc_library(
+    name = "heap5",
+    srcs = ["FreeRTOS/Source/portable/MemMang/heap_5.c"]
+)
+
+cc_library(
+    name = "posix",
+    srcs = [
+        "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/wait_for_event.c",
+        "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/wait_for_event.h",
+        "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/port.c",
+        "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/portmacro.h"
+    ]
+)
