--- /dev/null
+++ a/BUILD.bazel
@@ -0,0 +1,38 @@
+
+cc_library(
+    name = "freertos",
+    srcs = glob([
+        "FreeRTOS/Source/*.c",
+        "FreeRTOS/Source/portable/MemMang/heap_3.c",
+    ]) + [
+        "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/wait_for_event.c",
+        "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/port.c",
+    ],
+    hdrs = glob(["FreeRTOS/Source/include/*.h"]),
+    strip_include_prefix = "FreeRTOS/Source/include/",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":freertos_config",
+        ":freertos_portmacro",
+        ":freertos_wait_for_event",
+    ],
+)
+
+cc_library(
+    name = "freertos_portmacro",
+    hdrs = ["FreeRTOS/Source/portable/ThirdParty/GCC/Posix/portmacro.h"],
+    strip_include_prefix = "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/",
+)
+
+cc_library(
+    name = "freertos_wait_for_event",
+    hdrs = ["FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/wait_for_event.h"],
+    strip_include_prefix = "FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/",
+)
+
+cc_library(
+    name = "freertos_config",
+    hdrs = [
+        "FreeRTOSConfig.h",
+    ],
+)