--- /dev/null
+++ a/BUILD.bazel
@@ -0,0 +1,35 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])  # BSD
+
+cc_library(
+    name = "libfluent",
+    srcs = [
+        "src/debug.h",
+        "src/emitter.cc",
+        "src/fluent.hpp",
+        "src/fluent/emitter.hpp",
+        "src/fluent/exception.hpp",
+        "src/fluent/logger.hpp",
+        "src/fluent/message.hpp",
+        "src/fluent/queue.hpp",
+        "src/fluent/socket.hpp",
+        "src/logger.cc",
+        "src/message.cc",
+        "src/queue.cc",
+        "src/socket.cc",
+    ],
+    hdrs = ["src/fluent.hpp"],
+    copts = [
+        "-fPIC",
+        "-pthread",
+    ],
+    includes = [
+        "src/",
+        "src/fluent/",
+    ],
+    deps = ["@msgpack"],
+    alwayslink = True,
+)
